langchain/sql_database.py

```diff
@@ -336,10 +329,18 @@ def run(self, command: str, fetch: str = "all") -> str:
                 if fetch == "all":
                     result = cursor.fetchall()
                 elif fetch == "one":
-                    result = cursor.fetchone()[0]  # type: ignore
+                    result = cursor.fetchone()  # type: ignore
                 else:
                     raise ValueError("Fetch parameter must be either 'one' or 'all'")
-                return str(result)
+
+                # Convert columns values to string to avoid issues with sqlalchmey
+                # trunacating text
+                if isinstance(result, list):
+                    return str(
+                        list(tuple(map(utils.truncate_word, row)) for row in result)
+                    )
+                else:
+                    return str(tuple(map(utils.truncate_word, result)))
```

Comment:
(style) This is a style comment so up to you whether you want to update or not. 

A lot of python engineers generally prefer comprehensions over `map` and `filter` even though these are present in the language.

https://www.artima.com/weblogs/viewpost.jsp?thread=98196

```suggestion
                    return str(tuple(utils.truncate_word(r) for r in result)
```